File: dart/high-scores/lib/high_scores.dart
class HighScores {
  List<int> scores;

  HighScores(this.scores) {

  }

  int latest() {
    return this.scores.last;
  }

  List _sortScores() {
    List sortedScores = List.generate(this.scores.length, (i) => 0);
    List.copyRange(sortedScores, 0, this.scores);
    sortedScores.sort();

    return sortedScores;
  }

  int personalBest() {
    return _sortScores().last;
  }

  List personalTopThree() {
    return _sortScores().reversed.toList().take(3).toList();
  }
}

File: csharp/high-scores/HighScores.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    private List<int> scores;

    public HighScores(List<int> list)
    {
        scores = list;
    }

    public List<int> Scores() => scores;

    public int Latest() => scores[^1];

    public int PersonalBest() => PersonalTopN(1).First();

    public List<int> PersonalTopThree() => PersonalTopN(3);

    private List<int> PersonalTopN(int n)
    {
        return scores.OrderDescending().Take(n).ToList();
    }
}